src/overlays/actors/ovl_En_Horse/z_en_horse.c

```diff
@@ -90,253 +139,3541 @@ static ColliderJntSphInit D_80A65F28 = {
+    EnHorse_RotateToPoint(this, globalCtx, &PLAYER->actor.world.pos, 400);
+    if (this->flags & ENHORSE_OBSTACLE) {
+        this->actor.world.rot.y += 800.0f;
+    }
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+}
+
+void EnHorse_Freeze(EnHorse* this) {
+    if (this->action != ENHORSE_ACT_CS_UPDATE && this->action != ENHORSE_ACT_HBA) {
+        if (sResetNoInput[this->actor.params] != 0 && this->actor.params != 4) {
+            this->noInputTimer = 0;
+            this->noInputTimerMax = 0;
+        }
+        this->prevAction = this->action;
+        this->action = ENHORSE_ACT_FROZEN;
+        this->cyl1.base.ocFlags1 &= ~OC1_ON;
+        this->cyl2.base.ocFlags1 &= ~OC1_ON;
+        this->jntSph.base.ocFlags1 &= ~OC1_ON;
+        this->animationIdx = ENHORSE_ANIM_IDLE;
+    }
+}
+
+void EnHorse_ChangeIdleAnimation(EnHorse* this, s32 arg1, f32 arg2);
+void EnHorse_StartMountedIdleResetAnim(EnHorse* this);
+void EnHorse_StartMountedIdle(EnHorse* this);
+void EnHorse_StartGalloping(EnHorse* this);
+
+void EnHorse_Frozen(EnHorse* this, GlobalContext* globalCtx) {
+    this->actor.speedXZ = 0.0f;
+    this->noInputTimer--;
+    if (this->noInputTimer < 0) {
+        this->cyl1.base.ocFlags1 |= OC1_ON;
+        this->cyl2.base.ocFlags1 |= OC1_ON;
+        this->jntSph.base.ocFlags1 |= OC1_ON;
+        if (this->playerControlled == 1) {
+            this->flags &= ~ENHORSE_FLAG_7;
+            if (this->actor.params == 4) {
+                EnHorse_StartMountedIdleResetAnim(this);
+            } else if (this->actor.params == 9) {
+                this->actor.params = 5;
+                if (globalCtx->csCtx.state != 0) {
+                    EnHorse_StartMountedIdle(this);
+                } else {
+                    this->actor.speedXZ = 8.0f;
+                    EnHorse_StartGalloping(this);
+                }
+            } else if (this->prevAction == 2) {
+                EnHorse_StartMountedIdle(this);
+            } else {
+                EnHorse_StartMountedIdleResetAnim(this);
+            }
+            if (this->actor.params != 0) {
+                this->actor.params = 0;
+                return;
+            }
+        } else {
+            if (this->prevAction == 5) {
+                EnHorse_ChangeIdleAnimation(this, 0, 0);
+                return;
+            }
+            if (this->prevAction == 6) {
+                EnHorse_ChangeIdleAnimation(this, 0, 0);
+                return;
+            }
+            EnHorse_ChangeIdleAnimation(this, 0, 0);
+        }
+    }
+}
+
+void EnHorse_StickDirection(Vec2f* curStick, f32* stickMag, s16* angle);
+
+void EnHorse_UpdateSpeed(EnHorse* this, GlobalContext* globalCtx, f32 brakeDecel, f32 brakeAngle, f32 minStickMag, f32 decel, f32 baseSpeed,
+                   s16 turnSpeed) {
+    s16* why_god;
+    f32 stickMag;
+    s16 stickAngle;
+    f32 temp_f12;
+    f32 traction;
+    s16 turn;
+    if (EnHorse_PlayerCanMove(this, globalCtx) == 0) {
```

Comment:
`!EnHorse_PlayerCanMove(this, globalCtx)` assuming this is treated as a bool (and any other occurances if relevant)